src/pages/subjects/Subjects.tsx

```diff
@@ -55,11 +56,22 @@ const Subjects = () => {
     []
   )
 
-  const { loading: subjectNamesLoading, response: subjectsNamesItems } =
-    useSubjectsNames({
-      category: categoryId,
-      transform
-    })
+  const {
+    loading: subjectNamesLoading,
+    response: subjectsNamesItems,
+    fetchData
+  } = useSubjectsNames({
+    fetchOnMount: false,
+    category: categoryId,
+    transform
+  })
+
+  useEffect(() => setIsFetched(false), [categoryId])
```

Comment:
you don`t need useeffect here, set it in func that updates category

Reply:
> you don`t need useeffect here, set it in func that updates category

thank you for your advice, removed it 👍